Comp2024/src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -284,7 +284,8 @@ public double getIntakePosition( )
 
   public boolean isNoteDetected( )
   {
-    return m_noteInIntake.get( );
+    final Debouncer m_withinTolerance = new Debouncer(2.0);
```

Comment:
You are very close here. First, if we create a "new" debouncer object each time we call isNoteDetected(), then the "memory" of the debouncer gets lost (recreated). So it won't debounce the signal. Moving the debouncer object to be a class member variable (something like m_noteDetected) would create the debouncer once and the memory would stay valid between callse to isNoteDetected(). (not sure what "final" is going to help in this case) Second, the parameter for creating the debouncer is the window of time it will debounce over, and you've specified 2.0 seconds. This should be a minimum of 2 periodic loops of the roborio which is ~20msec--so maybe 0.030 seconds.